### ISSUE TITLE
Digest not working with multiple instances of component

### ISSUE THREAD

**Issue by levilindsey** [angular/angular.js#16117]
**I'm submitting a ...**
- [x] bug report
- [ ] feature request
- [ ] other (Please do not submit support requests here (see above))

**Current behavior:**
AngularJS doesn't seem to detect changes on a directive I've written. It looks like it's working as expected when I include only a single instance of the element, but when I include more than one, my changes don't seem to take effect.

**Minimal reproduction of the problem with instructions:**
https://plnkr.co/edit/P3rOipT7XvNfyJQzc6Fh?p=preview

**Angular version:** 1.6.4

**Browser:** all

**Comment by frederikprijck** [comment]
Hi, your issue reads like a support question. This github repo is only for bug reports and feature requests. Please try one of the listed [Support Channels](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#question). Thanks!

----

In your plunkr, you are only hiding the first panel

```
panelElement = wrapperElement.querySelector('.panel-1');
panelCtrl = angular.element(panelElement).controller('fooPanel');
...
panelCtrl.isFooShown = false;
```

Seems to work as expected if you also access the second panel's controller: https://plnkr.co/edit/m8KJZeqZ69RJfJKHHmzU?p=preview

**Comment by Narretz** [comment]
Hi, @frederikprijck is right, your issue reads like a support question. This github repo is only for bug reports and feature requests. Please try one of the listed [Support Channels](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#question). Thanks!

**Comment by levilindsey** [comment]
@frederikprijck, yes, it works if we hide both panels. But it needs to be able to hide one panel independently of the other.

How should I have phrased this to not read as a support question? I'm not asking how to do something, or why my code is wrong. AngularJS is definitely broken and doing something wrong here.

**Comment by levilindsey** [comment]
I guess I'll create a new issue, and re-phrase the description so it's more obvious that AngularJS has a bug.

I filed https://github.com/angular/angular.js/issues/16122.

**Comment by frederikprijck** [comment]
@levilindsey It's not about rephrasing, sorry for the confusing about that. I don't think recreating a new issue was neccesary if you want to elaborate on the same matter (or at least there wasn't something wrong with the current issue's description).
I also hope I didn't sound rude by saying it reads as a support questions even tho u are only trying to help in reporting a bug.

If I get this correctly, it's being resolved (or well, explained) in #16122.

Even tho we're trying to be helpful (me and @gkalpak are also availble on places like gitter for these kind of questions/help) It's a good idea to try and keep this github repo clean from issues caused by incorrect usage / expectations, and tend to help those people with their problems in the appropriate place.

**Comment by levilindsey** [comment]
@frederikprijck Thanks for the follow-up. Sorry, about the noise!
